clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -532,15 +532,21 @@ void pollSelectionKeys(Set<SelectionKey> selectionKeys,
                     try {
                         channel.prepare();
                     } catch (AuthenticationException e) {
-                        if (channel.successfulAuthentications() == 0)
+                        boolean isReauthentication = channel.successfulAuthentications() > 0;
+                        if (!isReauthentication)
                             sensors.failedAuthentication.record();
                         else
                             sensors.failedReauthentication.record();
```

Comment:
Can we swap the `if/else` so that we don't have to negate the first `if`? That seems unnecessarily confusing.